### PR TITLE
feat(sui-domain): subscribe to domain Use Case without decorators

### PR DESCRIPTION
**Pull request by midudev**
- [x] Remove the need of a manual decorator to enable this feature. 🧽
- [x] Improve domain predictability. 💭 
- [x] Avoid having to download the chunk for subscribing to the useCase! ⚡
- [x] Allow a simpler API surface for the domain: 🏄‍♂️

```
domain.get('generate_search_url_search_use_case').execute...
domain.get('generate_search_url_search_use_case').subscribe...
```

How to use:
```js
domain.get('generate_search_url_search_use_case').subscribe(({params, error, result}) => {
  // doSomething
})
```

How to unsubscribe:
```js
const unsubscribe = domain.get('generate_search_url_search_use_case').subscribe(handleSubscription)
unsubscribe()
```

Unsubscribe on React useEffect:
```js
useEffect(() => {
    const {unsubscribe} = domain.get('generate_search_url_search_use_case').subscribe(handleSubscription)
    return unsubscribe
  }, [deps])
```

### CODE REVIEW

**Comment by desko27** [comment]
😮😮😮 This is the way

**Review by desko27 — APPROVED** [review]

**Review by adriadoz — APPROVED** [review]

**Review by javiauso — APPROVED** [review]

**Review by rmoralp — APPROVED** [review]

File: packages/sui-domain/src/EntryPointFactory.js [hunk 1]
```diff
@@ -42,13 +43,34 @@ export default ({useCases, config}) =>
...
+                  )
+                  return Promise.reject(e)
+                })
+            },
+            subscribe: callback => {
+              this.subscribers[key] = this.subscribers[key] || []
+              this.subscribers[key].push(callback)
+
+              // return a way to remove the listener
+              return () => this.subscribers[key].filter(l => l !== callback)
```
Comment (rmoralp) [inline packages/sui-domain/src/EntryPointFactory.js:73]: As a subjective opinion, I like more an `rxjs` API where you can get a subscription with a `unsubscribe` method.
Getting directly the `unsubscribe` function when `subscribe` has been executed can cause confusion to me.
Some `rxjs` docs: https://rxjs-dev.firebaseapp.com/guide/subscription

**Review by rmoralp — APPROVED** [review]
Great job!

File: packages/sui-domain/src/EntryPointFactory.js (hunk 1 above)
Comment (midudev) [reply packages/sui-domain/src/EntryPointFactory.js:73]: Then we could do something like this. The drawback I see is that we will lose the possibility of cleaning the effect without storing a variable.

```
useEffect(() => {
    const {unsubscribe} = domain.get('generate_search_url_search_use_case').subscribe()
// ...
    return unsubscribe
  }, [])
```

**Review by oriolpuig — APPROVED** [review]

**Review by nucliweb — APPROVED** [review]

**Review by andresz1 — APPROVED** [review]

**Review by quinwacca — APPROVED** [review]

**Review by AgonisticKatai — APPROVED** [review]

**Review by nucliweb — APPROVED** [review]

**Review by danivm — APPROVED** [review]

**Review by kikoruiz — APPROVED** [review]
💚

File: packages/sui-domain/README.md [hunk 2]
```diff
@@ -101,3 +101,17 @@ export default class CurrentUserService extends Service {
   }
 }
 ```
+
+## Listen a useCase
+
+`@s-ui/domain` includes a way to subscribe to eveyr useCase execution without the need of using any kind of decorator or external dependency.
```
Comment (jcger) [inline packages/sui-domain/README.md:107]: typo every

**Review by jcger — APPROVED** [review]

File: packages/sui-domain/README.md (hunk 2 above)
Comment (midudev) [reply packages/sui-domain/README.md:107]: Thanks!

**Pull request merged**